pkg/azurestore/azureservice.go

```diff
@@ -67,87 +72,92 @@ type AzBlob interface {
 }
 
 type BlockBlob struct {
-	Blob       *azblob.BlockBlobURL
-	AccessTier azblob.AccessTierType
-	Indexes    []int
+	Blob           *blockblob.Client
+	Indexes        []int
+	BlobAccessTier *blob.AccessTier
 }
 
 type InfoBlob struct {
-	Blob *azblob.BlockBlobURL
+	Blob *blockblob.Client
 }
 
 // New Azure service for communication to Azure BlockBlob Storage API
 func NewAzureService(config *AzConfig) (AzService, error) {
 	// struct to store your credentials.
-	credential, err := azblob.NewSharedKeyCredential(config.AccountName, config.AccountKey)
+	cred, err := azblob.NewSharedKeyCredential(config.AccountName, config.AccountKey)
 	if err != nil {
 		return nil, err
 	}
 
-	// Might be limited by the storage account
-	// "" or default inherits the access type from the Storage Account
-	var containerAccessType azblob.PublicAccessType
+	serviceURL := fmt.Sprintf("%s/%s", config.Endpoint, config.ContainerName)
+	retryOpts := policy.RetryOptions{
+		MaxRetries:    20,   // Max 20 retries
+		RetryDelay:    100,  // Retry after 100ms initially
+		MaxRetryDelay: 5000, // Max retry delay 5 seconds
+	}
+	containerClient, err := container.NewClientWithSharedKeyCredential(serviceURL, cred, &container.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Retry: retryOpts,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var containerAccessType container.PublicAccessType
 	switch config.ContainerAccessType {
 	case "container":
-		containerAccessType = azblob.PublicAccessContainer
+		containerAccessType = container.PublicAccessTypeContainer
 	case "blob":
-		containerAccessType = azblob.PublicAccessBlob
-	case "":
+		containerAccessType = container.PublicAccessTypeBlob
 	default:
-		containerAccessType = azblob.PublicAccessNone
+		containerAccessType = ""
```

Comment:
The previous default was `azblob.PublicAccessNone`. Was this change intentional?

Reply:
there is no such type anymore, it is just the default behaviour if createOptions is nil. I did refactor according to this